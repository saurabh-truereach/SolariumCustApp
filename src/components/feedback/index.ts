/**
 * Feedback Components Export
 */

export {default as LoadingOverlay} from './LoadingOverlay';
export {default as ErrorBoundary} from './ErrorBoundary';
