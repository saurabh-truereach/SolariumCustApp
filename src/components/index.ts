/**
 * Components Export
 */

export * from './layout';
export * from './feedback';
