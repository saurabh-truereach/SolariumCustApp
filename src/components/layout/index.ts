/**
 * Layout Components Export
 */

export {default as SafeAreaLayout} from './SafeAreaLayout';
