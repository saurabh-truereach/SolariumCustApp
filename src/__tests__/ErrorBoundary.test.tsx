/**
 * Error Boundary tests
 */

import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import {ErrorBoundary} from '../components';

const ThrowError = ({shouldThrow}: {shouldThrow: boolean}) => {
  if (shouldThrow) {
    throw new Error('Test error');
  }
  return <Text>No error</Text>;
};

describe('ErrorBoundary', () => {
  it('renders children when there is no error', () => {
    const tree = renderer.create(
      <ErrorBoundary>
        <ThrowError shouldThrow={false} />
      </ErrorBoundary>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('catches and displays error', () => {
    const tree = renderer.create(
      <ErrorBoundary>
        <ThrowError shouldThrow={true} />
      </ErrorBoundary>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('calls onError callback when error occurs', () => {
    const onError = jest.fn();
    
    renderer.create(
      <ErrorBoundary onError={onError}>
        <ThrowError shouldThrow={true} />
      </ErrorBoundary>
    );
    
    expect(onError).toHaveBeenCalled();
  });
});
