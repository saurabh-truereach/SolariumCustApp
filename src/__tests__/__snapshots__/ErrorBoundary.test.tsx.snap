// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary catches and displays error 1`] = `null`;

exports[`ErrorBoundary renders children when there is no error 1`] = `null`;
